thentos-tests/src/Thentos/Test/Core.hs

```diff
@@ -208,14 +212,19 @@ setupTestServerFull = do
 
     return $ FTS tcfg asg backend beConfig frontend feConfig wd
 
-teardownTestServerFull :: FTS -> IO ()
+teardownTestServerFull :: forall db . (db `Extends` DB, IsAcidic db, Eq (ThentosError db)) =>
+    FTS db -> IO ()
 teardownTestServerFull (FTS tcfg db backend _ frontend _ _) = do
     cancel backend
     cancel frontend
     teardownDB $ DBTS tcfg db
 
 
-loginAsGod :: ActionState DB -> IO (ThentosSessionToken, [Header])
+loginAsGod :: forall db .
+        ( db `Extends` DB , IsAcidic db
+        , db ~ DB  -- FIXME https://github.com/liqd/thentos/issues/193
```

Comment:
We could, but I wanted to write down as much of the *intended* structure as possible, and I also want to fix this next, so this line should not stay there for long.  So I'd like to leave it as is.